### PR TITLE
[DOCS] Document async SQL APIs

### LINKED ISSUE #74845: [DOCS] SQL: Document async SQL search
We need to add documentation for new async SQL search functionality and APIs added with #73991.

### PR DESCRIPTION
**Pull request by jrodewig**
Adds formal API docs and JSON specs for the following APIs:

* Get async SQL search
* Get async SQL search status
* Delete async SQL search

I plan to migrate some other existing APIs (SQL search, translate, clear cursor) to this page in a separate PR.

Closes #74845

### Previews
* SQL APIs: https://elasticsearch_75078.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/sql-apis.html
* Get async SQL search: https://elasticsearch_75078.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/get-async-sql-search-api.html
* Get async SQL search status: https://elasticsearch_75078.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/get-async-sql-search-status-api.html
* Delete async SQL search: https://elasticsearch_75078.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/delete-async-sql-search-api.html

### CODE REVIEW

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
@elasticmachine update branch

File: docs/reference/sql/apis/get-async-sql-search-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+==== {api-path-parms-title}
+
+`<search_id>`::
+(Required, string) Identifier for the search.
+
+[[get-async-sql-search-api-query-params]]
+==== {api-query-parms-title}
+
+`delimiter`::
+(Optional, string) Separator for CSV results. Defaults to `,`. The API only
```
Comment (bpintea) [inline docs/reference/sql/apis/get-async-sql-search-api.asciidoc:44]: 👍

File: rest-api-spec/src/main/resources/rest-api-spec/api/sql.get.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{
```
Comment (bpintea) [inline rest-api-spec/src/main/resources/rest-api-spec/api/sql.get.json:1]: It should be clear that this is different from `sql.query.json`, but don't know if clear enough that all these are `async` SQL APIs. Up to you, not sure if there's a conversion about this.

**Review by bpintea — APPROVED** [review]
LGTM, thank you.

**Comment by stevejgordon** [comment]
LGTM!

File: rest-api-spec/src/main/resources/rest-api-spec/api/sql.get.json (hunk 2 above)
Comment (jrodewig) [reply rest-api-spec/src/main/resources/rest-api-spec/api/sql.get.json:1]: Thanks @bpintea. I used the same naming scheme we have for EQL:

* [eql.delete.json](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/eql.delete.json)
* [eql.get_status.json](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/eql.get_status.json)
* [eql.get.json](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/eql.get.json)

However, I'll add `async` to the SQL versions so its a bit more clear.

**Comment by jrodewig** [comment]
@elasticmachine ok to test

**Pull request merged**